contracts/xasset/src/test.rs

```diff
@@ -183,56 +248,196 @@ fn test_liquidation() {
-}
\ No newline at end of file
+    token.mint(&bob, &1200_0000000);
+    token.stake(&bob, &100_0000000);
+    let result = token.try_withdraw(&bob, &200_0000000);
+    assert!(result.is_err());
+}
+
+#[test]
+fn test_cdp_operations_with_interest() {
+    let e = Env::default();
+    e.mock_all_auths();
+
+    let xlm_admin_address = Address::generate(&e);
+    let (sac_contract, xlm_admin) = create_sac_token_clients(&e, &xlm_admin_address);
+    let xlm_token_address = xlm_admin.address.clone();
+    let datafeed = create_data_feed(&e);
+    let admin: Address = Address::generate(&e);
+    let token = create_token_contract(&e, admin, datafeed, xlm_token_address);
+
+    let alice = Address::generate(&e);
+    xlm_admin.mint(&alice, &2_000_000_000_000);
+
+    // Mock prices
+    let xlm_contract = token.xlm_contract();
+    let client = data_feed::Client::new(&e, &xlm_contract);
+    let xlm_price = 10_000_000_000_000;
+    client.set_asset_price(&Asset::Other(Symbol::new(&e, "XLM")), &xlm_price, &1000);
+
+    let usdt_contract = token.asset_contract();
+    let client = data_feed::Client::new(&e, &usdt_contract);
+    let usdt_price: i128 = 100_000_000_000_000;
+    client.set_asset_price(&Asset::Other(Symbol::new(&e, "USDT")), &usdt_price, &1000);
+
+    // Set initial timestamp
+    let initial_time = 1700000000;
+    loam_sdk::soroban_sdk::testutils::Ledger::set_timestamp(&e.ledger(), initial_time);
+
+    // Open initial CDP
+    token.open_cdp(&alice, &10_000_000_000, &500_000_000);
+    let initial_cdp = token.cdp(&alice);
+    assert_eq!(initial_cdp.xlm_deposited, 10_000_000_000);
+    assert_eq!(initial_cdp.asset_lent, 500_000_000);
+    assert_eq!(initial_cdp.accrued_interest.amount, 0);
+
+    // Advance time by 1 year (31536000 seconds)
+    loam_sdk::soroban_sdk::testutils::Ledger::set_timestamp(&e.ledger(), initial_time + 31536000);
+
+    // Check interest has accrued (11% annual rate)
+    let cdp_after_year = token.cdp(&alice);
+    assert!(cdp_after_year.accrued_interest.amount > 0);
+    // With 11% interest rate, expect ~55_000_000 interest (500_000_000 * 0.11)
+    assert!(cdp_after_year.accrued_interest.amount >= 54_000_000); // Allow for some rounding
+
+    // Advance another 6 months
+    loam_sdk::soroban_sdk::testutils::Ledger::set_timestamp(&e.ledger(), initial_time + 47304000);
+
+    // Borrow more
+    token.borrow_xasset(&alice, &200_000_000);
+
+    // Advance 3 more months
+    loam_sdk::soroban_sdk::testutils::Ledger::set_timestamp(&e.ledger(), initial_time + 55944000);
+
+    // Check total debt (original + borrowed + accumulated interest)
+    let cdp_before_repay = token.cdp(&alice);
+    assert!(cdp_before_repay.asset_lent + cdp_before_repay.accrued_interest.amount > 700_000_000);
+
+    // Use contract-provided view to get approval info
+    let interest_detail = token.get_accrued_interest(&alice);
+    let approval_amount = interest_detail.approvalAmount; // amount of xlm needed for the next 5 minutes
```

Comment:
If the struct keys are changed to snake case, then I think you can deconstruct this from the initial call more easily. Something like
```suggestion
    let InterestDetail { approval_amount, .. } = token.get_accrued_interest(&alice); // amount of xlm needed for the next 5 minutes
```